content/200-orm/200-prisma-client/100-queries/063-excluding-fields.mdx

```diff
@@ -4,15 +4,158 @@ metaTitle: 'Excluding fields'
+
+<TabbedContent code>
+
+<TabItem value="Schema">
+
+```prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["omitApi"]
+}
+
+model User {
+  id        Int      @id @default(autoincrement())
+  firstName String
+  lastName  String
+  email     String   @unique
+  password  String
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+}
+```
+
+</TabItem>
+
+<TabItem value="Code">
+
+```tsx
+const prisma = new PrismaClient({
+  omit: {
+    user: {
+      password: true
+    }
+  }
+})
+
+// The password field is excluded in all queries, including this one
+const user = await prisma.user.findUnique({ where: { id: 1 } })
+```
+
+</TabItem>
+
+</TabbedContent>
+
+## Excluding a field locally using `omit`
+
+The following is a type-safe way to exclude a field locally using the `omitApi` Preview feature:
+
+<TabbedContent code>
+
+<TabItem value="Schema">
+
+```prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["omitApi"]
+}
+
+model User {
+  id        Int      @id @default(autoincrement())
+  firstName String
+  lastName  String
+  email     String   @unique
+  password  String
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+}
+```
+
+</TabItem>
+
+<TabItem value="Code">
+
+```tsx
+const prisma = new PrismaClient()
+
+// The password field is excluded only in this query
+const user = await prisma.user.findUnique({
+  omit: {
+    password: true
+  },
+  where: { 
+    id: 1 
+  } 
+})
+```
+
+</TabItem>
+
+</TabbedContent>
+
+## How to omit multiple fields
+
+Omitting multiple fields works the same as selecting multiple fields: add multiple key-value pairs to the omit option.
+Using the same schema as before, you could omit password and email with the following:
 
-</TopBlock>
+```tsx
+const prisma = new PrismaClient()
+
+// password and email are excluded
+const user = await prisma.user.findUnique({
+  omit: {
+    email: true,
+    password: true,
+  },
+  where: {
+    id: 1,
+  },
+})
+```
+
+Multiple fields can be omitted locally and globally.
+
+## How to select a previously omitted field
+
+If you [omit a field globally](#excluding-a-field-globally-using-omit), you can select it by setting `omit` to `false` in any query.
+
+```tsx
+const user = await prisma.user.findUnique({
+  omit: {
+    password: false // The password field is now selected.
+  },
+  where: {
+    id: 1
+  }
+})
+```
 
-## Excluding the password field
+## When to use `omit` globally or locally
+
+It's important to understand when to omit a field globally or locally:
+
+- If you are omitting a field due to security reasons, it's best to omit it globally. For example: Globally omitting the password field from a User model so that sensitive information doesn't accidentally get exposed.
+- If you are omitting a field because it's not needed in a query, it's best to omit it locally.
```

Comment:
```suggestion
It's important to understand when to omit a field globally or locally:

- If you are omitting a field due to security reasons, it's best to omit it globally. For example: Globally omitting the password field from a User model so that sensitive information doesn't accidentally get exposed.
- If you are omitting a field because it's not needed in a query, it's best to omit it locally.

Locally omitted fields are omitted at the query level. This means that what fields are omitted can differ per-query, but there is no "default" omitted fields. On the other hand, globally omitted fields are defined at when a Prisma Client is instantiated and are then automatically omitted from every subsequent query on that Client. 

To omit different fields or select previous omitted fields, you will need to instantiate a new Prisma Client or [override a global omit with a local select](#how-to-select-a-previously-omitted-field).

```